.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,227 @@
+name: GEOS CI 
+on: pull_request
+
+# Cancels in-progress workflows for a PR when updated
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  GEOSX_TPL_TAG: 220-932
+
+jobs:
+  # Matrix jobs will be cancelled if PR is a draft.
+  # PR status must be "Open" to run CI.
+  check_pull_request_is_not_a_draft:
+    # We use the most recent ubuntu distribution available in Github Actions to ensure maximum support of google cloud's sdk.
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check that the PR is not a draft (cancel rest of jobs otherwise)
+      run: |
+        echo "Is PR a draft?"
+        echo ${{ toJSON(github.event.pull_request.draft) }}
+        if [[ ${{ toJSON(github.event.pull_request.draft) }} == true ]]; then "false" ; else "true"; fi
+
+  # PR must be assigned to be merged.
+  # This job will fail if this is not the case.
+  check_pull_request_is_assigned:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-latest
+    steps:
+    # Assignee ID will be null if PR is not assigned
+    - name: Check that the PR is assigned
+      run: |
+        id=$(curl -H "Accept: application/vnd.github+json" \
+           https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.number }} \
+           | jq '.assignee.id')
+        echo "Is PR Assigned? (Check for Assignee ID)"
+        echo $id
+        if [[ $id != null ]]; then "true" ; else "false"; fi
+
+  check_submodules:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-latest
+    steps:
+    # The integrated test sumbmodule repository contains large data (using git lfs).
+    # To save time (and money) we do not let Github Actions automatically clone all our (lfs) subrepositories and do it by hand.
```

Comment:
It looks like there's a `lfs` option that we can set to `false` to prevent us from doing the submodule clone by hand. (in `ci_build_and_test.sh` and/or `test_submodule_updated.sh`)